### PR TITLE
Prototyping multiple markdown files per pattern variant

### PR DESCRIPTION
**Pull request by jesstelford**
Styling is a bit wonky, and will pose some challenges unless we either add more markup to the markdown, or simplify the page layout...

localhost_3000_patterns_date-picking_tab=date-list (1)

### CODE REVIEW

File: polaris.shopify.com/content/patterns/date-picking/index.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+description: Makes it easy for merchants to select and input dates and date ranges.
+url: /patterns/date-picking
+previewImg: /images/patterns/pattern-thumbnail-date-picking.png
+status: {value: 'Beta', message: ''}
+order: 10
+githubDiscussionsLink: https://github.com/Shopify/polaris/discussions/7852
+variants:
+  - 'variants/single-date.md'
+  - 'variants/date-range.md'
+  - 'variants/date-list.md'
```
Comment (alex-page) [inline polaris.shopify.com/content/patterns/date-picking/index.md:11]: This is 1000x better then everything in one file IMO. Lots of ways to do this though.

**Comment by jesstelford** [comment]
Have gone through and matched the latest designs & `main`. Have also rebased onto `main` to bring in the latest functionality.

**Review by gwyneplaine — APPROVED** [review]

**Pull request merged**